src/components/common/alert-dialog.vue

```diff
@@ -35,7 +41,14 @@ withDefaults(
           </AlertDialog.AlertDialogTitle>
 
           <AlertDialog.AlertDialogDescription v-if="description">
-            {{ description }}
+            <template v-if="typeof description === 'object'">
+              {{ description.message }}
+              <strong>{{ description.boldText }}</strong>
+              {{ description.details }}
+            </template>
+            <template v-else>
+              {{ description }}
+            </template>
```

Comment:
Use Vue's slots instead of this. Current implementation is hard to maintain and extend, and yet it doesn't allow to pass icons, buttons, etc in the future.